tedana/interfaces/tedana.py

```diff
@@ -1365,165 +1343,214 @@ def writeresults(OCcatd, comptable, mmix, nt, acc, rej, midk, empty, head):
+
+    os.chdir(out_dir)
 
     print("++ Computing Mask")
     global mask
     mask, masksum = makeadmask(catd, minimum=False, getsum=True)
 
     print("++ Computing T2* map")
-    global t2s, s0, t2ss, s0s, t2sG, s0G
-    t2s, s0, t2ss, s0s, t2sG, s0G = t2sadmap(catd, mask, tes, masksum, 1)
+    global t2s, s0, t2sG
+    t2s, s0, t2ss, s0s, t2sG, s0G = t2sadmap(catd, mask, tes, masksum,
+                                             start_echo=1)
 
     # Condition values
     cap_t2s = stats.scoreatpercentile(t2s.flatten(), 99.5,
                                       interpolation_method='lower')
     t2s[t2s > cap_t2s*10] = cap_t2s
-    niwrite(s0, aff, 's0v.nii', head)
-    niwrite(t2s, aff, 't2sv.nii', head)
-    niwrite(t2ss, aff, 't2ss.nii', head)
-    niwrite(s0s, aff, 's0vs.nii', head)
-    niwrite(s0G, aff, 's0vG.nii', head)
-    niwrite(t2sG, aff, 't2svG.nii', head)
+    niwrite(s0, aff, op.join(out_dir, 's0v.nii'), head)
+    niwrite(t2s, aff, op.join(out_dir, 't2sv.nii'), head)
+    niwrite(t2ss, aff, op.join(out_dir, 't2ss.nii'), head)
+    niwrite(s0s, aff, op.join(out_dir, 's0vs.nii'), head)
+    niwrite(s0G, aff, op.join(out_dir, 's0vG.nii'), head)
+    niwrite(t2sG, aff, op.join(out_dir, 't2svG.nii'), head)
 
     # Optimally combine data
-    combmode = options.combmode
     global OCcatd
-    OCcatd = optcom(catd, t2sG, tes, mask,
-                    combmode, useG=True)
-    if not options.no_gscontrol:
-        catd, OCcatd = gscontrol_raw(OCcatd, head, len(tes))
+    OCcatd = optcom(catd, t2sG, tes, mask, combmode, useG=True)
+    if not no_gscontrol:
+        catd, OCcatd = gscontrol_raw(OCcatd, head, n_echoes)
 
-    if options.mixm is None:
+    if mixm is None:
         print("++ Doing ME-PCA and ME-ICA")
-
-        nc, dd = tedpca(combmode, mask, stabilize, head, ste=options.ste)
-
-        mmix_orig = tedica(nc, dd, options.conv, options.fixed_seed,
-                           cost=options.initcost,
-                           final_cost=options.finalcost)
-        np.savetxt('__meica_mix.1D', mmix_orig)
+        n_components, dd = tedpca(combmode, mask, stabilize, head, tes=tes, kdaw=kdaw,
+                        rdaw=rdaw, ste=ste)
+        mmix_orig = tedica(n_components, dd, conv, fixed_seed, cost=initcost,
+                           final_cost=finalcost)
+        np.savetxt(op.join(out_dir, '__meica_mix.1D'), mmix_orig)
         seldict, comptable, betas, mmix = fitmodels_direct(catd, mmix_orig,
                                                            mask, t2s, t2sG,
                                                            tes, combmode, head,
-                                                           fout=options.fout,
+                                                           fout=fout,
                                                            reindex=True)
-
-        np.savetxt('meica_mix.1D', mmix)
+        np.savetxt(op.join(out_dir, 'meica_mix.1D'), mmix)
 
         if 'GROUP0' in sys.argv:
```

Comment:
What do you think of getting rid of `group0` here and in `selcomps` ? I'm ok to remove the option since it was undocumented behavior, anyway. We can open an issue about documenting and adding it back in eventually, but I'd like to get rid of this `sys.argv` call sooner than later.

Reply:
I think that's a great idea. I'll remove it posthaste.